src/biome/text/pipeline.py

```diff
@@ -179,18 +188,16 @@ def find_lr(
         """
         from biome.text._helpers import create_trainer_for_finding_lr
 
-        if vocabulary.is_empty(
-            self._model.vocab, self.config.features.keys
-        ):
+        if vocabulary.is_empty(self._model.vocab, self.config.features.keys):
             raise EmptyVocabError(
                 "Found an empty vocabulary. "
                 "You probably forgot to create a vocabulary with '.create_vocabulary()'."
             )
 
-        if isinstance(training_data, DataSource):
-            training_data = self.create_dataset(training_data)
-        elif isinstance(training_data, Dataset):
-            training_data: AllennlpLazyDataset = training_data.to_instances(pipeline=self)
+        if isinstance(training_data, Dataset):
+            training_data: AllennlpLazyDataset = training_data.to_instances(
+                pipeline=self
+            )
```

Comment:
The default value of `lazy` has changed, so we have to set it to True here.
Nothing to do with the removal of `DataSource`, but let's include it in this PR as well, since i just saw it.
```suggestion
            training_data: AllennlpLazyDataset = training_data.to_instances(
                pipeline=self, 
                lazy=True,
            )
```